### PR TITLE
[gmime] Update to 3.2.15, use official build system

### PR DESCRIPTION
**Pull request by dg0yt**
Stop maintaining the vendored build system which is incomplete even after #42898.

FTR upstream development was always on github, cf. https://github.com/jstedfast/gmime/issues/119. New releases are only there.

We cannot bootstrap the buildsystem from the on-the-fly tarballs, so we must use the official release distfiles.

### CODE REVIEW

File: versions/g-/gmime.json [hunk 1]
```diff
@@ -1,5 +1,15 @@
 {
   "versions": [
+    {
+      "git-tree": "9c6ff1ec537247679019f8f25826423dd63d0f2c",
+      "version": "3.2.15",
+      "port-version": 0
+    },
+    {
+      "git-tree": "5b1f053f75ec146c2f4cd22afbe62f5d4673bc20",
```
Comment (Thomas1664) [inline versions/g-/gmime.json:9]: One version is enough

Comment (dg0yt) [reply versions/g-/gmime.json:9]: Yeah, but there must be some food for review comments ;-)

File: versions/g-/gmime.json [hunk 2]
```diff
@@ -1,5 +1,15 @@
...
+      "version": "3.2.6",
+      "port-version": 7
```
Comment (dg0yt) [inline versions/g-/gmime.json:11]:
```suggestion
```

File: ports/gmime/vcpkg.json [hunk 3]
```diff
@@ -1,19 +1,18 @@
...
+  "supports": "!xbox",
   "dependencies": [
     "glib",
-    "libiconv",
-    "libidn2",
     {
-      "name": "vcpkg-cmake",
+      "name": "gmime",
       "host": true
     },
```
Comment (dg0yt) [inline ports/gmime/vcpkg.json:13]: The host dependency is for reusing `iconv-detect.h` at least for the same cmake system name. Alternatively we could add some files to the port dir. Windows, osx and linux results saved in https://gist.github.com/dg0yt/8451f384696a84ba6d2e70c475e93d0f.

**Review by Mengna-Li — APPROVED** [review]

**Pull request merged**